### PR TITLE
Add Configuration Converter in otx.tools

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

https://jira.devtools.intel.com/browse/CVS-134576
- Add the Converter to `otx.tools`.
- Modify the functions in Converter to be static.
- Improve readability of some parameter update code.
- Add an `instantiate` function to Converter that creates a dict with engine and train arguments based on a configuration of type dict.

Converter is now available as shown below.
```python
from otx.tools.converter import ConfigConverter

geti_det_config = "tests/assets/geti-configs/det.json"
data_root = "tests/assets/car_tree_bug"
work_dir = "otx-workspace-geti"


config = ConfigConverter.convert(geti_det_config)
engine, train_kwargs = ConfigConverter.instantiate(
    config=config,
    data_root=data_root,
    work_dir=work_dir,
)

engine.train(**train_kwargs)
```

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/tools/converter.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    },
+    "Custom_Counting_Instance_Segmentation_MaskRCNN_ResNet50": {
+        "task": OTXTaskType.INSTANCE_SEGMENTATION,
+        "model_name": "maskrcnn_r50",
+    },
+    "Custom_Semantic_Segmentation_Lite-HRNet-18-mod2_OCR": {
+        "task": OTXTaskType.SEMANTIC_SEGMENTATION,
+        "model_name": "litehrnet_18",
+    },
+}
```
Comment (jaegukhyun) [inline src/otx/tools/converter.py]: This dictionary should be able to cover all of Geti models

**Comment by harimkang** [comment]
@vinnamkim @jaegukhyun 
I found some issue with multi-label, h-label classification,

In OTX 1.X (1.6), The template is used without distinction between multi class classification, multi label classification, and h label classification (They share one template.), and the model is determined at runtime via the functions below.

https://github.com/openvinotoolkit/training_extensions/blob/620dfd1c45387da7ad5d2ecd005219414620d4d1/src/otx/algorithms/classification/task.py#L134-L157

https://github.com/openvinotoolkit/training_extensions/blob/620dfd1c45387da7ad5d2ecd005219414620d4d1/src/otx/algorithms/classification/adapters/mmcls/task.py#L92-L106

We need to discuss how to handle this in Geti. I think we should get the relevant task information from Geti or Dataset.

File: src/otx/tools/converter.py (hunk 1 above)
Comment (harimkang) [reply src/otx/tools/converter.py]: Thanks! Done - https://github.com/openvinotoolkit/training_extensions/pull/3254/commits/ae6f4077c583684ffac6afb5edcca93adce00803

**Comment by vinnamkim** [comment]
> We need to discuss how to handle this in Geti. I think we should get the relevant task information from Geti or Dataset.

See https://github.com/intel-innersource/frameworks.ai.interactive-ai-workflow.sonoma-creek/blob/433e1c46f123033b559935739552a9e1c6c9fc4a/src/workload_configurations/otx2/scripts/definitions.py#L99-L105
You can assume that `task_type` will be given

**Comment by harimkang** [comment]
> > We need to discuss how to handle this in Geti. I think we should get the relevant task information from Geti or Dataset.
> 
> See https://github.com/intel-innersource/frameworks.ai.interactive-ai-workflow.sonoma-creek/blob/433e1c46f123033b559935739552a9e1c6c9fc4a/src/workload_configurations/otx2/scripts/definitions.py#L99-L105 You can assume that `task_type` will be given

Thanks for letting me know.

**Review by jaegukhyun — APPROVED** [review]

**Review by vinnamkim — APPROVED** [review]

**Pull request merged**